python/dgl/nn/mxnet/conv.py

```diff
@@ -0,0 +1,158 @@
+"""MXNet modules for graph convolutions."""
+# pylint: disable= no-member, arguments-differ
+import mxnet as mx
+from mxnet import gluon
+
+from ... import function as fn
+
+__all__ = ['GraphConv']
+
+class GraphConv(gluon.Block):
+    r"""Apply graph convolution over an input signal.
+
+    Graph convolution is introduced in `GCN <https://arxiv.org/abs/1609.02907>`__
+    and can be described as below:
+
+    .. math::
+      h_i^{(l+1)} = \sigma(b^{(l)} + \sum_{j\in\mathcal{N}(i)}\frac{1}{c_{ij}}W^{(l)}h_j^{(l)})
+
+    where :math:`\mathcal{N}(i)` is the neighbor set of node :math:`i`. :math:`c_{ij}` is equal
+    to the product of the square root of node degrees:
+    :math:`\sqrt{|\mathcal{N}(i)|}\sqrt{|\mathcal{N}(j)|}`. :math:`\sigma` is an activation
+    function.
+
+    The model parameters are initialized as in the
+    `original implementation <https://github.com/tkipf/gcn/blob/master/gcn/layers.py>`__ where
+    the weight :math:`W^{(l)}` is initialized using Glorot uniform initialization
+    and the bias is initialized to be zero.
+
+    Notes
+    -----
+    Zero in degree nodes could lead to invalid normalizer. A common practice
+    to avoid this is to add a self-loop for each node in the graph, which
+    can be achieved by:
+
+    >>> g = ... # some DGLGraph
+    >>> g.add_edges(g.nodes(), g.nodes())
+
+
+    Parameters
+    ----------
+    in_feats : int
+        Number of input features.
+    out_feats : int
+        Number of output features.
+    norm : bool, optional
+        If True, the normalizer :math:`c_{ij}` is applied. Default: ``True``.
+    dropout: float, optional
+        The probability of setting an element in node feature to be zero before performing
+        graph convolution during training. When it is 0, no dropout is performed. Default: ``0``.
+    bias : bool, optional
+        If True, adds a learnable bias to the output. Default: ``True``.
+    activation: callable activation function/layer or None, optional
+        If not None, applies an activation function to the updated node features.
+        Default: ``mxnet.nd.relu``.
+    feat_name : str, optional
+        The temporary feature name used to compute message passing. Default: ``"_gconv_feat"``.
+
+    Attributes
+    ----------
+    weight : mxnet.gluon.parameter.Parameter
+        The learnable weight tensor.
+    bias : mxnet.gluon.parameter.Parameter
+        The learnable bias tensor.
+    """
+    def __init__(self,
+                 in_feats,
+                 out_feats,
+                 norm=True,
+                 dropout=0.,
+                 bias=False,
+                 activation=mx.nd.relu,
+                 feat_name="_gconv_feat"):
```

Comment:
Let's not put dropout and activation here. My idea is to make it similar to torch.nn.conv2d. Activation and dropout can be handled outside.

Reply:
My motivation is to make this more like Kipf's initial implementation of [GraphConvolutionLayer](https://github.com/tkipf/gcn/blob/master/gcn/layers.py#L132) where both dropout and activation are part of it. This also make potential model code more compact and expressive like `nn.Linear(bias=True)` is more compact than performing matrix multiplication, check whether bias is to be added and then perform a bias addition if needed.

Reply:
Ok, to me this is a more [Tensorflow/Keras way](https://keras.io/layers/core/), but I'm think it's not a big deal. One thing is that you might want to set `activation=None` by default, which is more commonly seen.

Reply:
Also, dropout is actually a tricky stuff. Where is the right place to apply it? On the input? On the aggregated value? Or after activation? I feel there is so many decisions so it's good to leave it to users. (TF/Keras also does not include dropout).

Reply:
Done.